kpatch-build/create-diff-object.c

```diff
@@ -1133,7 +1149,8 @@ static void kpatch_replace_sections_syms(struct kpatch_elf *kelf)
 			 * bail out.  If we hit this situation, more core is
 			 * needed here to calculate the value of 'add_off'.
 			 */
-			if (rela->type == R_PPC64_REL24)
+			if (rela->type == R_PPC64_REL24 &&
+			    strcmp(sec->name, ".rela.fixup"))
 				ERROR("Unexpected relocation type R_PPC64_REL24 for %s\n", rela->sym->name);
```

Comment:
Is this ERROR even needed?  I think 'add_off' is an x86-specific quirk.  It seems like it will always be 0 on ppc64le.

Reply:
The initial assumption was that all `R_PPC64_REL24`  symbols are functions and should be replaced with their respective section symbol. There are cases where the `R_PPC64_REL24` function relocates to `section + offset`. In short for the latter we need to handle it more like bundled section and yes this check can be removed.